RandLAPACK/comps/rl_util.hh

```diff
@@ -402,6 +585,24 @@ T cond_num_check(
     return cond_num;
 }
 
+// Computes the numerical rank of a given matirx
+template <typename T>
+int64_t rank_check(
+    int64_t m,
+    int64_t n,
+    const std::vector<T>& A
+) {
+    std::vector<T> A_pre_cpy;
+    std::vector<T> s;
+    RandLAPACK::util::cond_num_check(m, n, A, A_pre_cpy, s, false);
+
+    for(int i = 0; i < n; ++i) {
+        if (s[i] / s[0] <= 5 * std::numeric_limits<double>::epsilon())
```

Comment:
Should have ``std::numeric_limits<T>::epsilon()``. You've currently hardcoded ``T = double``.

Reply:
Done.